torch/nn/parameter.py

```diff
@@ -52,10 +52,15 @@ def __new__(cls, data=None, requires_grad=True):
     def __deepcopy__(self, memo):
         if id(self) in memo:
             return memo[id(self)]
+
+        if _get_torch_function_mode():
```

Comment:
I don't think that's the way to go there, this should just properly implement the `__torch_function__` handler (similar to the Tensor `__deepcopy__`).
cc @jbschlosser

Reply:
How do you mean ?

Reply:
I think he means do this instead of referencing modes:
```python
if has_torch_function_unary(self):
    return handle_torch_function(Tensor.__deepcopy__, (self,), self, memo)
```

However, it looks like this is unneeded in the most recent code update?

Reply:
TODO for us to fix `Parameter.__deepcopy__()` to call `self.data.__deepcopy__()` instead of `self.data.clone()`. The two implementations have different view-preserving behavior because of how they interact with the memo.